src/content/effective-dart/design.md

```diff
@@ -621,41 +620,23 @@ doesn't do that, it's best to assume you should *not* extend the class.
 Otherwise, later changes to it may break your code.
 
 
-### DO document if your class supports being extended
+### DO use class modifiers to control if your class can be extended
 
-This is the corollary to the above rule. If you want to allow subclasses of your
-class, state that. Suffix the class name with `Base`, or mention it in the
-class's doc comment.
+Class modifiers like `final`, `base`, or `sealed` 
+allow you to enforce whether a class can be extended directly in code. 
+For example, use `final class A {}` to prevent extension, 
+or `base class B {}` to allow extension only within the same library. 
+Rely on these modifiers rather than documentation to communicate and 
+enforce your intent.
 
 
-### AVOID implementing a class that isn't intended to be an interface
+### ### DO use class modifiers to control if your class can be an interface
```

Comment:
Your proposed rule is from the perspective of someone writing the library, but the original version was intended for those who are using a library. This section is talking about classs that aren't intended to be an interface, and while it would be nice if all such classes were using class modifiers to enforce these rules (as recommended above), it's still possible that a developer could run into these problems so I would vote to keep this section as-is.